src/traces/sunburst/attributes.js

```diff
@@ -0,0 +1,145 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var plotAttrs = require('../../plots/attributes');
+var hovertemplateAttrs = require('../../components/fx/hovertemplate_attributes');
+var domainAttrs = require('../../plots/domain').attributes;
+var pieAtts = require('../pie/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+// TODO should we use singular `label`, `parent` and `value`?
```

Comment:
For `sankey` traces, we went with singular attribute names. Maybe we should stick to that? 

What do @nicolaskruchten, @alexcjohnson @plotly/plotly_js think?

Reply:
I favour singular everywhere

Reply:
So, should we change the `pie` attributes `labels` and `values` accordingly?

Reply:
... and `marker.colors`

Reply:
...of course I favour consistency above all!

Reply:
> So, should we change the pie attributes labels and values accordingly?
> ... and marker.colors

Ok so it seems like we have more plurals than singulars.

> ...of course I favour consistency above all!

Me too so I guess we should stick with what we already have right now (ie. plural form). Maybe in the future, we could support both.

Reply:
At one point I think the idea was `arrayOk` attributes - which could be scalars OR arrays - should be singular, and `data_array` attributes should be plural. But I don't think we were ever consistent about that, and anyway it doesn't work to pluralize things like `x`. So yeah, if I were to redo it all today I'd probably vote 100% singular (and perhaps we should make that change in a major bump?), but at this point wherever we're using a name that's already in the schema somewhere we should try to match it.

Reply:
> but at this point wherever we're using a name that's already in the schema somewhere we should try to match it.

those were my thoughts exactly :ok_hand:

Reply:
:+1: I guess you can :zap: that TODO line and add it to our wishlist for v2

Reply:
TODO :zap: in https://github.com/plotly/plotly.js/pull/3594/commits/1258f2da606954952b7e2bfc1c82eb7a274b2ef1

added to v2 list:

![image](https://user-images.githubusercontent.com/6675409/55171175-3fba8c80-514e-11e9-90ba-1c0b42e36b17.png)